src/sage/categories/finite_monoids.py

```diff
@@ -220,6 +220,24 @@ def rhodes_radical_congruence(self, base_ring=None):
                         res.append((m, n))
             return res
 
+        @cached_method
+        def one(self):
+            """
+            Return the unit.
+
+            This is the generic implementation, very naive and slow.
+
+            EXAMPLES::
+
+                sage: R = IntegerModRing(15)
+                sage: M = R.subsemigroup([R(5)], one=R(10),
+                ....:     category=Semigroups().Finite().Subobjects() & Groups())
+                sage: M.one()
+                10
+            """
+            it = (v for v in self if all(v * w == w for w in self))
```

Comment:
```suggestion
            """
            try:
                return super().one()
            except Exception:
                pass
            it = (v for v in self if all(v * w == w for w in self))
```
I think this is necessary to take care of the failures.